libs/testserver/fake_workspace.go

```diff
@@ -82,18 +83,18 @@ func (s *FakeWorkspace) WorkspaceDelete(path string, recursive bool) {
 	}
 }
 
-func (s *FakeWorkspace) WorkspaceFilesImportFile(path string, body []byte) {
-	if !strings.HasPrefix(path, "/") {
-		path = "/" + path
+func (s *FakeWorkspace) WorkspaceFilesImportFile(filePath string, body []byte) {
+	if !strings.HasPrefix(filePath, "/") {
+		filePath = "/" + filePath
 	}
-	s.files[path] = body
+	s.files[filePath] = body
 
 	// Add all directories in the path to the directories map
-	parts := strings.Split(path, "/")
+	parts := strings.Split(path.Dir(filePath), "/")
```

Comment:
What remains odd is that this routine first splits and then rebuilds the path.

You can call `path.Dir` in a loop until the result is "/" and stick intermediate values in the map for the same result. I didn't test below, but it illustrates what I'm thinking of.

```go
	for dir := path.Dir(path); dir != "/"; dir = path.Dir(dir) {
		s.directories[dir] = true
	}
```